### ISSUE TITLE
`object-property-newline` with enhanced object literals

### ISSUE THREAD

**Issue by kevva** [eslint/eslint#6211]
Using the `object-property-newline` rule in conjunction with an object containing only enhanced object literals written on the same line reports as an error, as expected. What I want is an option to allow for writing the props on the same line _if_ it contains only enhanced object literals. Since those are defined outside the object it makes it kinda moot to write them on different lines.

![](https://cloud.githubusercontent.com/assets/709159/15292945/e75e045a-1b85-11e6-8996-9fabef3626be.png)

**Comment by platinumazure** [comment]
I'd love to see more examples of both good and bad code here. I'm not following the proposal as initially stated.

**Comment by eslintbot** [comment]
Hi @kevva, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by kevva** [comment]
Bad:

```js
const a = 'foo';
const b = 'bar';

fn({
	a,
	b
});

fn({a, b, unicorn: true});
```

Good:

```js
const a = 'foo';
const b = 'bar';

fn({a, b});

fn({
	a,
	b,
	unicorn: true
});
```

**Comment by nzakas** [comment]
I think what you're saying is that if you are using shorthand initialization for properties, that it should not require new lines?

Note: it helps us a lot to follow our contribution guidelines (http://eslint.org/docs/developer-guide/contributing/rule-changes).

**Comment by kevva** [comment]
Yeah, exactly :). Or at least having an opttion to allow it for objects containing only shorthands.

**Comment by nzakas** [comment]
@eslint/eslint-team thoughts on this?

@kevva are you willing to implement this if we accept this proposal?

**Comment by kevva** [comment]
@nzakas, sure, no problem.

**Comment by pedrottimark** [comment]
@kevva It looks like the rule already has an option http://eslint.org/docs/rules/object-property-newline#allowmultiplepropertiesperline to allow all on one line.

Is it a problem for you that it does not apply only to shorthand initializations?

**Comment by kevva** [comment]
@pedrottimark, I only want objects containing only shorthands to be allowed on one line.

**Comment by nzakas** [comment]
It's up to the team, not me specifically.

**Comment by ilyavolodin** [comment]
Anyone on the @eslint/eslint-team wants to champion this enhancement?

**Comment by mikesherov** [comment]
I'm :+1: and would potentially champion once I close some other issues.

**Comment by mysticatea** [comment]
My thought is #6251.
I thought this and other JSCS rules in total, in that issue.

**Comment by kaicataldo** [comment]
I'm closing this issues because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
